pkg/envoy/registry/types.go

```diff
@@ -14,9 +15,13 @@ var log = logger.New("proxy-registry")
 type ProxyRegistry struct {
 	ProxyServiceMapper
 
-	connectedProxies sync.Map
+	connectedProxies map[string]*envoy.Proxy
 
 	msgBroker *messaging.Broker
+
+	proxiesCount int
```

Comment:
standard practice is to call it `mu sync.Mutex`

and simply place the protected fields underneath it, in this case the map itself. No need to store the length, you can just len whenever you need it. that's a fast op

Reply:
+1